brigade-controller/Dockerfile

```diff
@@ -1,7 +1,16 @@
-FROM alpine:3.8
+FROM quay.io/deis/lightweight-docker-go:v0.6.0
+ARG LDFLAGS
+ENV CGO_ENABLED=0
+WORKDIR /go/src/github.com/brigadecore/brigade
+COPY brigade-controller/ brigade-controller/
+COPY pkg/ pkg/
+COPY vendor/ vendor/
+RUN go build -ldflags "$LDFLAGS" -o bin/brigade-controller ./brigade-controller/cmd/brigade-controller
 
-RUN apk add --no-cache ca-certificates && update-ca-certificates
-
-COPY rootfs/brigade-controller /usr/bin/brigade-controller
-
-CMD /usr/bin/brigade-controller
+FROM scratch
```

Comment:
I think we may need to bring in a `/tmp` dir for this image as well (as is done in `brigade-vacuum`'s).

As it stands, I'm seeing the ctlr pod CrashLoopBackOff w/:
```
 $ k logs brigade-brigade-ctrl-7c5495577d-kkq2v
W0531 16:21:32.534596       1 client_config.go:533] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
log: exiting because of error: log: cannot create log: open /tmp/brigade-controller.brigade-brigade-ctrl-7c5495577d-kkq2v.unknownuser.log.WARNING.20190531-162132.1: no such file or directory
```

Reply:
@vdice, this is what the size of those dirs looks like:

```
156M	bin/
128K	brig/
 16K	brigade-api/
 72K	brigade-controller/
 20K	brigade-cr-gateway/
 24K	brigade-generic-gateway/
 28K	brigade-vacuum/
153M	brigade-worker/
3.6M	docs/
 32K	git-sidecar/
476K	pkg/
 20K	scripts/
 12K	tests/
101M	vendor/
```

Your idea would work, but would drag along the very large `vendor/` dir even for images that don't involve any Go code being built-- e.g. the worker. I think maybe what we can do it `.gitignore` `.dockerignore` (for good measure) and then copy an image-specific `.dockerignore` from each component's directory over to the root of the project right before the build is kicked off.

Regardless of your strategy or mine, one minor complication will be that building multiple images cannot be parallelized because there will be a race over the contents of `.dockerignore` in the root or the project. Of course, this is a bridge we could cross if and when we come to it. `make build-all-images` builds images serially right now.

Reply:
> I think we may need to bring in a /tmp dir for this image as well (as is done in brigade-vacuum).

Thought I did. I will check on that.

Reply:
> I think we may need to bring in a /tmp dir for this image as well (as is done in brigade-vacuum).

@vdice not sure what happened that you are seeing that error. I've double-checked that the new image works. The `--logtostderr` seems to be honored by glog, so writes to `/tmp` aren't coming into play.

Can you double check that this is still happening to you?

Reply:
I also just pushed some new commits and one of them reduces the build context for each image, as you suggested @vdice

Reply:
Very nice -- the build context solution added in https://github.com/brigadecore/brigade/pull/920/commits/ce72e8f738662e1695b6f0c6883d7aa6ce18768e is _much_ more elegant than my suggestion above.  👍